kernel/kernel-api/src/test/scala/io/delta/kernel/CloseableIteratorSuite.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel
+
+import scala.collection.JavaConverters._
+
+import io.delta.kernel.internal.util.Utils
+import io.delta.kernel.utils.CloseableIterator
+import io.delta.kernel.utils.CloseableIterator.BreakableFilterResult
+import org.scalatest.funsuite.AnyFunSuite
+
+class CloseableIteratorSuite extends AnyFunSuite {
+
+  private def toCloseableIter[T](elems: Seq[T]): CloseableIterator[T] = {
+    Utils.toCloseableIterator(elems.iterator.asJava)
+  }
+
+  private def toList[T](iter: CloseableIterator[T]): List[T] = {
+    iter.toInMemoryList.asScala.toList
+  }
+
+  private def normalDataIter = toCloseableIter(Seq(1, 2, 3, 4, 5))
+
+  private def throwingDataIter = toCloseableIter(Seq(1, 2, 3, 4, 5)).map { x =>
+    if (x > 4) {
+      throw new RuntimeException("Underlying data evaluated at element > 4")
+    }
+    x
+  }
+
+  test("CloseableIterator::filter -- returns filtered result") {
```

Comment:
add a case where nothing is filtered and we reach the end of the iterator? I think none of these tests reach the end

Reply:
it can be for any of the methods

Reply:
Doesn't this case here reach that? This is just a normal filter -- we will iterate over all of the elements in the list

Reply:
sorry they all reach the end; the last element is never returned though; just seems slightly different conceptually

Reply:
Ah, now I understand. For sure, will add.